### DISCUSSION TITLE
Did I download the wrong WIX version?

### DISCUSSION THREAD

**Discussion by LuvWhiteSand** [wixtoolset/issues#6430] [Questions]
Downloaded WIX to windows 10 tried to start it but it says .net framework 3.5.x needs to be installed.   I found this question in numerous places but no real answer.  Found a link that says WIX 3.14 will fix this but that was 4 years ago.  Is this really still not fixed?

**Comment by robmen** [comment]
WiX v3.11 and older requires .NET Framework v3.5 because backwards compatibility is its primary goal. You can install .NET Framework v3.5 on Windows 10 using "Turn Windows features on or off".

WiX v4 moves up to .NET Core v3.1 and .NET Framework v4.6.1+.

WiX v3.14 is a "stepping stone release" that has breaking changes in it aimed to help the migration from v3.x to v4.0 and as such no longer supports .NET Framework v3.5 (like v4.0).

**Reply by LuvWhiteSand** [reply]
So I am brand new to using WiX  is there a link for WiX v4

**Reply by LuvWhiteSand** [reply]
Ok,  I found the v4  don't know why I couldn't find it a couple of days ago.

**Reply by Evangelink** [reply]
Where is version 4 available? The wix release page only shows v3.11.2 (https://wixtoolset.org/releases/)

**Reply by robmen** [reply]
WiX v4 is still under development. You can find ancient early builds in the ["Development Builds" page](https://wixtoolset.org/releases/development/) but those are not indicative of what WiX v4 looks like now. WiX v4-preview.0 is coming out very, very soon.

**Reply by Evangelink** [reply]
Oh ok then I will stick with v3.11.2 for now. Thanks for the quick reply and the explanation.

**Reply by jbogari** [reply]
Rob,

Similarly, I'm trying to author a new bundle using the WiX 3.11 provided BAL. Attempting to run the produced bundle on target systems without .NET 3.5 present seem to fail at the .NET version detection stage and the bundle install crashes. I gather this focus on backward compatibility might also lead to the assumption that .NET 3.5 will always be present?

Is there some way to overcome this mismatch without telling every potential user they have to install .NET 3.5 on newer systems that don't have it installed (Windows 11, Server 2022) in order to run a bundle from 3.11?

Thanks.

Jeff

**Reply by robmen** [reply]
> I gather this focus on backward compatibility might also lead to the assumption that .NET 3.5 will always be present?

Nope.

> Is there some way to overcome this mismatch without telling every potential user they have to install .NET 3.5 on newer systems that don't have it installed (Windows 11, Server 2022) in order to run a bundle from 3.11?

Yep. Your bundle controls what .NET Framework it requires.

PS: Your question isn't related to the original question, so please don't continue in this thread.
